debugger/src/debug_adapter/protocol.rs

```diff
@@ -63,11 +63,21 @@ impl<R: Read, W: Write> DapAdapter<R, W> {
         let response_header = format!("Content-Length: {}\r\n\r\n", response_body.len());
 
         self.output.write_all(response_header.as_bytes())?;
+        self.output.flush()?;
 
-        if let Err(err) = self.output.write_all(response_body) {
-            // TODO: Implement catch of errorkind "WouldBlock" and retry
-            tracing::error!("send_data - body: {:?}", err);
-            return Err(err);
+        // NOTE: Sometimes when writing large response, the debugger will fail with an IO error (ErrorKind::WouldBlock == error.kind())
```

Comment:
If this is an issue, I would suggest not using `write_all`. The `write` function returns the number of bytes written, and then that can be called in a loop until all data is written.

Reply:
Actually, that was the first thing I tried and it works on smaller payloads, but with a large payload (in one of the user's C testcases, it returns a 48K static array), when I used `write()` the response data got corrupted and I'd get VSCode hanging.  What I have here (break the write into chuncks) is the only reliable way I found to transfer that response to the client.